Mathlib/LinearAlgebra/BilinearForm/Basic.lean

```diff
@@ -265,8 +265,8 @@ theorem flip_flip :
 
 /-- The `flip` of a bilinear form over a commutative ring, obtained by exchanging the left and
 right arguments. -/
-abbrev flip : BilinForm R M ≃ₗ[R] BilinForm R M :=
-  flipHom
+abbrev flip (B : BilinForm R M) :=
```

Comment:
You mentioned running into issues with `flip`. I'm not sure what these were but I'll comment that this change is necessary for dot notation (which you use elsewhere) to work.

The way dot notation works is _roughly_ that if you write `foo.bar` and there is no lemma with that name, Lean looks for a lemma named `X.bar` where `X` is the type of `foo` and which has an argument of type `X`. If it finds this, it then elaborates `foo.bar` as `X.bar foo`. So in this case because `LinearEquiv` is not a function (merely something with a coercion to a function) `flip` will not support dot notation until this change is made.

Reply:
Looking more closely I think I can see what was the issue. I believe the import you added to `Lie.Killing` allowed Lean find `LinearMap.BilinForm.flip` whereas previously it could not. Since `LinearMap.BilinForm` is an `abbrev`, previously when it failed to find `LinearMap.BilinForm.flip` it then looked for, any found, `LinearMap.flip` and all was well (since this has an argument of the right type). However once you added the import, Lean was able to find `LinearMap.BilinForm.flip` and all was not well until you made the change to its definition.

Reply:
Thanks. It makes sense! Things indeed started to be messed up as soon as I imported a file. But when fiddling around, I worked under the assumption that all existing code was perfectly good, so I was reluctant to making the change.

By the way, I just pushed the penultimate step there: #12106